### PR TITLE
General improvements to generated documentation

### PR DESCRIPTION
**Pull request by slimsag**
- Keep comments in both `event` and `elem` packages at readable lengths (wrap at 80 chars).
- Start func docstrings with the func name itself (following Go style conventions).
- Document name mapping properly (/cc @shurcooL).

### CODE REVIEW

**Comment by shurcooL** [comment]
This looks incredible! Thanks for doing it.

> Start func docstrings with the func name itself (following Go style conventions).

Are you looking to cover all cases, or just the vast majority? I see some entries that would still trigger golint warnings, would you want me to point them out?

**Comment by slimsag** [comment]
> Are you looking to cover all cases, or just the vast majority? I see some entries that would still trigger golint warnings, would you want me to point them out?

Correct, there are still a large number that do cause golint warnings at this point (I would estimate 20-30% but don't have a real number on hand); These will mostly be solved by updating the MDN and regenerating -- which can be done independently of this PR. I will follow-up on these :)

File: event/generate.go [hunk 1]
```diff
@@ -138,15 +145,20 @@ func main() {
...
+		e.Spec = strings.TrimSpace(cols.Eq(2).Text())
 
 		funName := nameMap[e.Name]
 		if funName == "" {
 			funName = capitalize(e.Name)
 		}
 
+		if e.Desc != "" {
+			// Lowercase the first letter of the description.
+			lowercaseDesc := strings.ToLower(string(e.Desc[0])) + e.Desc[1:]
```
Comment (shurcooL) [inline event/generate.go]: It seems inconsistent that `capitalize` is also used in just one place, yet it's made into a helper, but the same isn't done for its lower equivalent.

Comment (shurcooL) [inline event/generate.go]: As a bonus, if you make it an unexported helper, then the comment above can be removed since it'd be unneccessary.

**Comment by shurcooL** [comment]
One minor comment. LGTM.

File: event/generate.go (hunk 1 above)
Comment (slimsag) [inline event/generate.go]: Very good suggestion, thanks for this! Done. :)

**Comment by slimsag** [comment]
Since there have been no further comments and I've addressed all by shurcooL, I'm going to merge. I think this change won't be very controversial, but @neelance if you see anything here you dislike or want to discuss we can revert that portion and I'll resubmit it as a PR for further discussion.

**Pull request merged**